### ISSUE TITLE
Streaming the request as well as the response?

### ISSUE THREAD

**Issue by stoarca** [jimhigson/oboe.js#61]
Hi there,

Are there any plans to make oboe stream the request as well as the response? Currently, oboe JSON.stringify'es the request body, which works for small requests. However,  for large requests (~150MB), it causes the node process to hang.

See this stackoverflow question for more discussion: http://stackoverflow.com/questions/28288896/node-js-posting-a-large-request-using-request-module-hangs

**Comment by jimhigson** [comment]
Yes, this is a good suggestion. What do you think an API would look like?

This is possible already in Node if you open the http request yourself and
pass the response stream into Oboe (rather than have Oboe make the request
on your behalf) - see BYO stream <http://oboejs.com/api#byo-stream>
